### ISSUE TITLE
[clang, c++, concepts] Cannot use _BitInt in type requirements

### ISSUE THREAD

**Issue by Matthewacon** [llvm/llvm-project#53057]
**Affected versions**: 12.0.0 and onward

**Example 1 (clang 14.0.0+)**:
```c++
template<auto N>
requires requires {
 typename _BitInt(N);
}
using T = void;
```

**Example 2 (clang versions 12.0.0 - 13.0.0)**:
```c++
template<auto N>
requires requires {
 typename _ExtInt(N);
}
using T = void;
```

**Diagnostic**:
```
<source>:3:11: error: expected a qualified name after 'typename'
 typename _BitInt(N);
          ^
1 error generated.
```
**Notes**:
Can work around the issue by using an alias template, as follows:
```c++
template<auto N>
using BitInt = _BitInt(N);

template<auto N>
requires requires {
 typename BitInt<N>;
}
using T = void;
```

**Comment by AaronBallman** [comment]
Can confirm this issue (why is there no way to put this information into metadata like through a label??).

**Comment by AaronBallman** [comment]
It's actually a bit worse than that -- there's a failed assertion involved: https://godbolt.org/z/Tjf37hd86

**Comment by AaronBallman** [comment]
The failed assertion is now fixed in fa596fb0779ae9029edbcff80ff95e9d1a816206.

As for accepting the code `typename _BitInt(N);`, I think we're correct to reject this. _BitInt(N) is a dependent type, so I can see why it'd be nice for it to work, but the C++ grammar expects a `type-name` and not a `simple-type-specifier`, and `_BitInt(N)` is closer to the latter than the former. I believe the "workaround" is the correct approach until `_BitInt` is brought into C++, where I would expect it to be spelled `std::bit_int<N>` and will work as expected.
